goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/pipeline.py

```diff
@@ -5,9 +5,13 @@
 
 from kedro.pipeline import Pipeline, pipeline, node
 from .nodes import apply_book_attributes
+from .nodes import merge_description_embeddings
+from .nodes import perform_clustering_analysis
 
 
 def create_pipeline(**kwargs) -> Pipeline:
```

Comment:
We should look to restructure the pipeline. Since the books_features target is already in the catalog and being used by other pipelines as input, I'd like to keep it as the last step for this pipeline to avoid a need to refactor elsewhere. I would rearrange this to start with clustering and then pass the dataset with the cluster on it already to `apply_book_attributes()`